### PR TITLE
#4691 - Identify performance problems in detailed export #4316 - SAMP…

### PR DESCRIPTION
**Pull request by cazacmarin**
…LES section - additional Tests

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java [hunk 1]
```diff
@@ -729,7 +727,8 @@ public void validate(SampleDto sample) throws ValidationRuntimeException {
 		Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight, I18nProperties.getCaption(Captions.inaccessibleValue));
 
 		for (SampleExportDto exportDto : resultList) {
-			List<PathogenTest> pathogenTests = sampleService.getById(exportDto.getId()).getPathogenTests();
+			Sample sample1 = sampleService.getById(exportDto.getId());
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:730]: please rename `sample1` to `sample`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java [hunk 2]
```diff
@@ -965,80 +976,125 @@ private void generateSamples() {
...
-			sample.setLab(config.getLaboratory());
+				sample.setLab(config.getLaboratory());
 
-			if (config.isRequestPathogenTestsToBePerformed()) {
-				Set set = new HashSet<PathogenTestType>();
-				int until = randomInt(1, PathogenTestType.values().length);
-				for (int j = 0; j < until; j++) {
-					set.add(PathogenTestType.values()[j]);
+				if (config.isRequestPathogenTestsToBePerformed()) {
+					Set set = new HashSet<PathogenTestType>();
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java:1010]: please rename `set` to `pathogetnTestTypes`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java [hunk 3]
```diff
@@ -965,80 +976,125 @@ private void generateSamples() {
...
-				sample.setRequestedPathogenTests(set);
-			}
 
-			if (config.isRequestAdditionalTestsToBePerformed()) {
-				Set set = new HashSet<AdditionalTestType>();
-				int until = randomInt(1, AdditionalTestType.values().length);
-				for (int j = 0; j < until; j++) {
-					set.add(AdditionalTestType.values()[j]);
+				if (config.isRequestAdditionalTestsToBePerformed()) {
+					Set set = new HashSet<AdditionalTestType>();
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java:1020]: please rename `set` to `additionalTestTypes`

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java (hunk 1 above)
Comment (cazacmarin) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:730]: Variable sample is already defined in that scope.

Comment (leventegal-she) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:730]: Then rename also the other one

**Review by leventegal-she — CHANGES_REQUESTED** [review]

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**